daemon/stats/collector.go

```diff
@@ -89,7 +90,10 @@ func (s *Collector) Run() {
 		for _, pair := range pairs {
 			stats, err := s.supervisor.GetContainerStats(pair.container)
 			if err != nil {
-				if _, ok := err.(notRunningErr); !ok {
+				// if we get "container not found" error from containerd, it's possibly
+				// because that this container has already been stopped. it will be ok to
+				// ignore this error and just return an empty stats.
+				if _, ok := err.(notRunningErr); !ok && !strings.Contains(err.Error(), "container not found") {
```

Comment:
Not introduced in this PR, but while looking at this code:
The double negative is a bit confusing here as well; would this be more readable (assuming a typed error for `container not found` was implemented;?

```go
for _, pair := range pairs {
	stats, err := s.supervisor.GetContainerStats(pair.container)

	switch err.(type) {
	case nil:
		// FIXME: move to containerd on Linux (not Windows)
		stats.CPUStats.SystemUsage = systemUsage
		stats.CPUStats.OnlineCPUs = onlineCPUs

		pair.publisher.Publish(*stats)

	case notRunningErr, notFoundErr:
		// publish empty stats containing only name and ID if not running or not found
		pair.publisher.Publish(types.StatsJSON{
			Name: pair.container.Name,
			ID:   pair.container.ID,
		})

	default:
		logrus.Errorf("collecting stats for %s: %v", pair.container.ID, err)
	}
}
```